### PR TITLE
[Validator] Specify attributes of fileType thanks to ValidatorTypeGuesser #29327

### PR DESCRIPTION
**Pull request by BorisCharriere**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes   
| Fixed tickets | #29327 
| License       | MIT
| Doc PR        | symfony

[Form] [Validator] 

this fix help the Validator to be more complete and user-friendly.  The Validator takes the value of the field ``` mimeTypes ``` and put it into the attribute of the FileType.

### CODE REVIEW

File: CHANGELOG-4.2.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ To get the diff between two versions, go to https://github.com/symfony/symfony/c
 
 * 4.2.0-BETA1 (2018-11-03)
 
+ * feature #29327 [Validator] Specify attributes of fileType thanks to validatorTypeGuesser
```
Comment (nicolas-grekas) [inline CHANGELOG-4.2.md]: this should be reverted: this file is auto-generated

File: UPGRADE-4.2.md [hunk 2]
```diff
@@ -373,6 +373,7 @@ Validator
  * The `checkMX` and `checkHost` options of the `Email` constraint are deprecated
  * The component is now decoupled from `symfony/translation` and uses `Symfony\Contracts\Translation\TranslatorInterface` instead
  * The `ValidatorBuilderInterface` has been deprecated and `ValidatorBuilder` made final
+ * The `ValidatorTypeGuesser` can now guess the type of accepted files and out it into the FileType
```
Comment (nicolas-grekas) [inline UPGRADE-4.2.md]: no need to change the UPGRADING file - but the CHANGELOG *of the component* should have that line (under 4.3.0)

**Review by nicolas-grekas — COMMENTED** [review]
thanks - can you add a test case, please?

**Comment by fabpot** [comment]
@BorisCharriere Still interested in finishing this PR?

**Comment by BorisCharriere** [comment]
@fabpot No, you can close it, I'll keep changes on my side and work on them when i'll have more time

**Pull request closed without merging**